### ISSUE TITLE
EditorState.set() behavior on handleReturn

### ISSUE THREAD

**Issue by hnordt** [facebook/draft-js#1161]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Using `EditorState.set()` on `handleReturn` doesn't look to behave correctly. 

Steps to reproduce: https://jsfiddle.net/7bsqzmfg/10/

**What is the expected behavior?**

Using `EditorState.set()` on handleReturn` should work just like `EditorState.push()`.

**Which versions of Draft.js, and which browser / OS are affected by this issue? Did this work in previous versions of Draft.js?**

Draft.js 0.10.0
Google Chrome 57.0.2987.133
macOS Sierra 10.12.3 (16D32)

**Comment by hellendag** [comment]
Yeah, the logic around setting selection is a little more specialized than `set`. I'm not really sure `EditorState.set` should be public, but since it is, the docs should be updated to note that selection setting should use selection-specific methods.
